spark/src/test/java/org/apache/iceberg/spark/source/TestSparkReaderDeletes.java

```diff
@@ -159,4 +168,72 @@ public void testEqualityDeleteWithFilter() throws IOException {
 
     Assert.assertEquals("Table should contain no rows", 0, actual.size());
   }
+
+  @Test
+  public void testReadDeleteRow() throws IOException {
+    String tableName = "testDeleteRowRead";
```

Comment:
I think we need an unit test to cover the case that have multiple version of equality delete schema to verify that the Predicates concatted by OR is working as expected.